net/instaweb/rewriter/csp.cc

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Author: morlovich@google.com (Maksim Orlovich)
+//
+// This provides basic parsing and evaluation of a (subset of)
+// Content-Security-Policy that's relevant for PageSpeed Automatic
+
+#include "net/instaweb/rewriter/public/csp.h"
+
+#include "net/instaweb/rewriter/public/csp_directive.h"
+
+namespace net_instaweb {
+
+namespace {
+
+void TrimCspWhitespace(StringPiece* input) {
+  // AKA RWS in HTTP spec, which of course isn't the HTML notion of whitespace
+  // that TrimWhitespace uses.
+  while (!input->empty() && ((*input)[0] == ' ' || (*input)[1] == '\t')) {
+    input->remove_prefix(1);
+  }
+
+  while (input->ends_with(" ") || input->ends_with("\t")) {
+    input->remove_suffix(1);
+  }
+}
+
+char Last(StringPiece input) {
+  DCHECK(!input.empty());
```

Comment:
should this be a CHECK?   Surely nothing good will come of this if input.empty() is true.